### PR TITLE
feat: add shadow color for inactive windows

### PR DESCRIPTION
**Pull request by eerii**
Solves #228.

This pull request adds a configuration option for changing the inactive windows' shadow color, `shadow_inactive_color`. It follows the same syntax as `shadow_color`. If it is not defined, it takes the default value of `shadow_color`, there is no change for existing configurations.

This is my first time contributing to this project and one of my first pull requests, so please if there is anything I have to do differently or change tell me. (Also thank you, I have been using this as my main desktop and couldn't be happier!)

https://github.com/WillPower3309/swayfx/assets/22449369/c7d83b78-73fa-4aa6-a1e0-406eb20c48a4

### CODE REVIEW

**Comment by AnotherRandomGitHubAccount** [comment]
Tested this out locally. can't find any bugs and works amazing! Thanks! can't wait for somebody to approve it

File: sway/commands/shadow_color.c
Comment (ErikReider) [inline sway/commands/shadow_color.c:21]: Why would you set the inactive color here?

Comment (eerii) [reply sway/commands/shadow_color.c:21]: I did it so if you only set a custom shadow_color, both of them are set to same one (same behaviour as now), but if you do set a shadow_inactive_color then they are separate. If you don’t set it there and only use shadow_color then shadow_inactive_color would still be the default. But I could find another place to do it if that is not fine.

Comment (ErikReider) [reply sway/commands/shadow_color.c:21]: I'd prefer if you check if it equals 0 instead of inverting the value. Also add curly brackets after the if-statement

Comment (eerii) [reply sway/commands/shadow_color.c]: Absolutely! Did you mean something like the new commit? Thanks

File: sway/desktop/render.c
Comment (ErikReider) [inline sway/desktop/render.c:890]: Keeping it in the command file is a better solution. Then this doesn't have to run every time we render something

**Review by ErikReider — CHANGES_REQUESTED** [review]

File: sway/commands/shadow_color.c
Comment (ErikReider) [reply sway/commands/shadow_color.c]: > Absolutely! Did you mean something like the new commit? Thanks

Yeah :)

File: sway/desktop/render.c
Comment (eerii) [reply sway/desktop/render.c]: Sorry, that makes a lot of sense. Should be fixed now

Comment (ErikReider) [reply sway/desktop/render.c]: > Sorry, that makes a lot of sense. Should be fixed now

Doesn't seem to work. The first `memcmp` implementation worked best. But as in that comment, check if the result equals 0 or not

Comment (eerii) [reply sway/desktop/render.c]: Ok, I'm so sorry. I completely misinterpreted what you meant by inverting the value and thought I had to do it another way. I feel a bit silly now ahaha. It's back to the first one but using equals zero and with brackets. Sorry for all the fuss!

**Review by ErikReider — COMMENTED** [review]
LGTM! I'll let @WillPower3309 merge this just in case he has any final thoughts :)

File: sway/commands/shadow_color.c
Comment (WillPower3309) [inline sway/commands/shadow_color.c:22]: I think this would result in shadow_inactive_color being set twice (once here and then again in the shadow_inactive_color command file), so this can be omitted

**Review by WillPower3309 — COMMENTED** [review]
LGTM, one nit

**Comment by WillPower3309** [comment]
This repo is pretty complex, great work as a first contributor!

**Comment by WillPower3309** [comment]
Feel free to ping me if you're on the SwayFX discord server for the contributor role

File: sway/commands/shadow_color.c
Comment (eerii) [reply sway/commands/shadow_color.c:22]: That is there so if you only set the shadow_color (and not the shadow_inactive_color) both of them are set to your custom one like right now, so that the changes are backward compatible. It does mean that it is set twice but the alternative is to make everyone change the configs to also add shadow_inactive_color even if they want shadows to be uniform. I'm ok changing it either way though, so please tell me!

**Comment by eerii** [comment]
> This repo is pretty complex, great work as a first contributor!

Thank you! ^^

File: sway/commands/shadow_color.c
Comment (eerii) [reply sway/commands/shadow_color.c:22]: Also if shadow_inactive_color is already set that is ignored, that's what the memcmp is for, so you can define them in any order.

Comment (WillPower3309) [reply sway/commands/shadow_color.c:22]: Ah got it, misinterpreted the comparison

**Review by WillPower3309 — APPROVED** [review]

**Pull request merged**
